app/src/main/java/com/iti/android/team1/ebuy/ui/cart_screen/view/CartFragment.kt

```diff
@@ -1,56 +1,100 @@
+    }
     private lateinit var cartProductAdapter: CartProductAdapter
     private val binding get() = _binding!!
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
+        savedInstanceState: Bundle?,
     ): View? {
         _binding = FragmentCartBinding.inflate(inflater, container, false)
         return binding.root
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        viewModel = ViewModelProvider(this).get(CartViewModel::class.java)
-        // TODO: Use the ViewModel
-    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         initCartRecycler()
+        handleAllCartItems()
+        handleCheckoutButton()
+        handleOverFlow()
+        handleDeleteState()
     }
 
-    // function
-    fun initCartRecycler() {
-        cartProductAdapter = CartProductAdapter().apply {
-            this.setCartProducts(
-                arrayListOf(
-                    CartProduct("Mouse", true, 6.0),
-                    CartProduct("Mouse", true, 6.0),
-                    CartProduct("Mouse", true, 6.0)
-                )
-            )
+    private fun handleDeleteState() {
+        viewModel.deleteState.observe(viewLifecycleOwner) { result ->
+            when (result) {
+                DatabaseResult.Empty -> {
+                    Toast.makeText(requireContext(),
+                        getString(R.string.delete_success),
+                        Toast.LENGTH_SHORT).show()
+                }
+                is DatabaseResult.Error -> {
+                    Toast.makeText(requireContext(),
+                        getString(R.string.delete_error),
+                        Toast.LENGTH_SHORT).show()
+                }
+
+            }
+
+        }
+    }
+
+    private fun handleCheckoutButton() {
+        binding.btnAddCard.setOnClickListener {
+            viewModel.updateToDB()
+        }
+    }
+
+    private fun handleAllCartItems() {
+        viewModel.getAllCartItems()
+        lifecycleScope.launch {
+            viewModel.allCartItems.observe(viewLifecycleOwner) { result ->
+                when (result) {
+                    ResultState.EmptyResult -> {
+                        cartProductAdapter.setCartItems(emptyList())
+                        handleTotalMoney(0L)
+                    }
+//                    is ResultState.Error -> TODO()
+//                    ResultState.Loading -> TODO()
+                    is ResultState.Success -> {
+                        cartProductAdapter.setCartItems(result.data)
+                        handleTotalMoney(result.data.sumOf {
+                            it.productVariantPrice * it.customerProductQuantity
+                        }.toLong())
+                    }
+                }
```

Comment:
what is the beneift of using lifeCycleScope here ?

Reply:
if we change live data with stateflow